docs/testnet-guardian-runbook.md

```diff
@@ -0,0 +1,130 @@
+
+##### Update brew package
+To update brew package,
+
+1. Download the homebrew binary package from gcloud at the location specified in the `Copy artifacts to cloud` task in the `build-macOS` CI job. For example `gs://network-debug/382998/build/homebrew-coda.tar.gz`
+2. Upload it to our public s3 bucket at `https://s3-us-west-2.amazonaws.com/packages.o1test.net`
+3. In `coda.rb` in `CodaProtocol/homebrew-coda`,
+        1. update the url with the location of the new tarball
+        2. update the hash. The file containing the hash is uploaded to gcloud as well, for example `gs://network-debug/382998/build/homebrew-coda.tar.gz.sha256`
+        3. increment the revision number
+
+##### Update apt package
+ 1. Create an annotated tag for the latest commit on develop after code freeze. For example: `git tag -m "release 0.0.13" 0.0.13`
+ 2. push the tag `git push origin 0.0.13`
+ 3. Push the changes from develop on to master
+
+##### Publish the location of the binaries
+In addition to updating the brew and apt packages, publish the locations to the binaries on the discord channel `#QA-net details` channel for the rest of the team
+1. macOS binary
+2. linux binary
+3. docker image
+4. Name of the release branch
+
+### Deploy
+
+Deploy a QA-net (currently a 200-node network) with the new binaries. Ensure gossip logs are enabled by setting the daemon flags `log-snark-work-gossip`, `log-txn-pool-gossip`, and `log-received-blocks`.
+
+Instructions for configuring and deploying a QA-net is roughly documented here: https://www.notion.so/codaprotocol/Testnet-deployment-notes-c1fbf8b8be9343b480f644c22d61ad85
+
+### Test
+
+After the QA-net is deployed, perform the following sanity-check:
+
+##### Sanity check
+
+1. Search for any crashes by querying the logs on StackDriver. The nodes are configured to restart upon crash for some of the known issues (#4295). Create issues for new ones.
+
+2. Run `scripts/ezcheck.sh` to check if all the nodes are synced and have the same state hash at their best tips.
+
+3. Query the logs to confirm that the nodes are creating and receiving blocks, transaction-pool diffs, and snark-pool diffs.
+Some strings to look for: `Received a block`, `Received Snark-pool diff`, `Received transaction-pool diff`
+
+4. Connect an external node to the network and check if it syncs
+(TODO: to automate these checks)
+
+Publish the peer ids of the seed nodes on `#QA-net details` channel for the rest of the team to connect to the QA-net.
+
+##### Complete the challenges for the release
+
+Connect to the QA-net from your machine and complete the protocol/product related challenges for the release. This is to test the features required for the challenges are working and also, may help the testnet guardian prepare for the kind of questions the community may have after the release.
```

Comment:
This is awesome!

Reply:
Some of the challenges may be "send 100 transactions" or "make the most snarks as a snark worker" -- these would translate to "send at least one transaction" and "make at least one snark work" for the guardian to test them out. Should we make this explicit?

Reply:
Putting on my devils advocate hat, it is probably beneficial to do the complete challenge. Might even be nice to produce code that does the challenge for you, committing it to a repo for later use in validation.

Reply:
@bkase @yourbuddyconner  I think it depends on the challenges. Makes sense to complete the "send n transactions" to see how the client responds or test out different ways it can be done (cli/graphql). But anything that is resource intensive like "make most snark works", it should ideally be part of the QA-net configuration to run multiple snark workers with different keys and fees and verify if they have been included. Alternatively, one could run a snark worker locally for a couple of hours and check how many of them were included?

Reply:
I think it depends on the challenges? "send n transactions" can be done in your local machine but not any resources intensive tasks like running snark worker/daemon for a long time? Such tests should ideally be part of the QA-net configuration. Or one could run it for a shorter duration and test it the way @bkase mentioned